tests/unit/test_simulation.py

```diff
@@ -265,6 +265,20 @@ def test_save_load(self):
         ):
             sim.save("test.pickle")
 
+    def test_load_param(self):
+        # Test load_sim for parameters imports
+        pkl_obj = pybamm.load_sim(
+            os.path.join(
+                "tests", "unit", "test_parameters", "data", "test_load_param.p"
```

Comment:
You committed the pickle file `test_load_param.p` but we don't really know what it corresponds to. How about, instead, this tests saves a simulation before trying to load it?

Reply:
@tlestang The bug is reproducible only when the pickle file is loaded afresh in a new terminal. If we'll save and then load it then the parameter will already be imported and the error won't occur.

Reply:
Ah yes. Alternatively we could write the pickle from another process, something like
```
save_sim = "import pybamm; model = pybamm.lithium_ion.SPM(); pybamm.Simulation(model).save("test_load_param.p")"
subprocess.run("python -c {save_sim}")
```
Relying on the binary pickle only may make debugging hard in the future, when nobody remembers what is contains :) 
Which makes me think another option is adding a comment that describes how you generated the pickle file.